crates/ra_assists/src/handlers/add_function.rs

```diff
@@ -224,12 +232,29 @@ fn fn_arg_name(fn_arg: &ast::Expr) -> Option<String> {
     }
 }
 
-fn fn_arg_type(ctx: &AssistCtx, fn_arg: &ast::Expr) -> Option<String> {
+fn fn_arg_type(
+    ctx: &AssistCtx,
+    target_module: &Option<hir::InFile<hir::ModuleSource>>,
+    fn_arg: &ast::Expr,
+) -> Option<String> {
     let ty = ctx.sema.type_of_expr(fn_arg)?;
     if ty.is_unknown() {
         return None;
     }
-    Some(ty.display(ctx.sema.db).to_string())
+    let target_syntax = match target_module {
+        None => fn_arg.syntax(),
+        Some(module) => match &module.value {
+            hir::ModuleSource::SourceFile(it) => it.syntax(),
+            hir::ModuleSource::Module(it) => it.syntax(),
+        },
+    };
+
+    let target_module = ctx.sema.scope(target_syntax).module()?;
```

Comment:
since you're only using `target_syntax` for this, you could also just get the module directly if `target_module` is `Some`

Reply:
I updated this part while trying to properly handle local imports, does this comment still apply?

Reply:
Oh, do you mean how I already have a `hir::Module` [here](https://github.com/rust-analyzer/rust-analyzer/blob/1a8d3dd2a0f9a476f28b0a900c225bf210df2525/crates/ra_assists/src/handlers/add_function.rs#L50)? True, there's some waste there, I think I know how to refactor this.